components/script/body.rs

```diff
@@ -72,6 +73,7 @@ struct TransmitBodyConnectHandler {
     bytes_sender: Option<IpcSender<Vec<u8>>>,
     control_sender: IpcSender<BodyChunkRequest>,
     in_memory: Option<Vec<u8>>,
+    in_memory_done: Cell<bool>,
```

Comment:
I think it can be just a `bool`, since you have a `&mut self` everywhere.

Reply:
Indeed